gazebo_state_plugins/src/GazeboObjectInfo.cpp

```diff
@@ -102,6 +125,85 @@ void GazeboObjectInfo::onWorldUpdate() {
     reGenerateObjects=false;
 }
 
+shape_msgs::Mesh * GazeboObjectInfo::getMesh(physics::CollisionPtr& c){
+    shape_msgs::Mesh tri;
+    msgs::Geometry geom;
+    physics::ShapePtr shape=c->GetShape();
+    shape->FillMsg(geom);
+
+    if(geom.has_mesh()){
+    	 //const gazebo::physics::MeshShape * mesh_shape = dynamic_cast<const physics::MeshShape*>(shape.get());
+       // const gazebo::physics::BoxShape * box=dynamic_cast<const physics::BoxShape*>(shape.get());
+
+    	boost::shared_ptr<gazebo::physics::MeshShape> mesh_shape = boost::dynamic_pointer_cast<gazebo::physics::MeshShape>(shape);
+    	 std::string name_ = mesh_shape->GetName();
+    	 std::string uri = mesh_shape->GetMeshURI();
+    	 const gazebo::common::Mesh *mesh = gazebo::common::MeshManager::Instance()->GetMesh(uri);
+         unsigned n_submeshes = mesh->GetSubMeshCount();
+
+         for(unsigned m=0; m < n_submeshes; m++) {
```

Comment:
I would try to use ++m instead of m++.
See for instance [this](http://stackoverflow.com/questions/24853/what-is-the-difference-between-i-and-i) and references therein.